File: POO/Heritage/todo-sequence2/humain.cpp
#include "humain.h"

// Constructeurs et Destructeur
Humain::Humain(const string nom/*=""*/, const string boissonFavorite/*="eau"*/) : nom(nom), boissonFavorite(boissonFavorite)
{
}

Humain::~Humain()
{
}

// Accesseurs
string Humain::getNom() const
{
   return nom;
}

string Humain::getBoissonFavorite() const
{
   return boissonFavorite;
}

void Humain::setBoissonFavorite(const string nouvelleBoissonFavorite)
{
   this->boissonFavorite = nouvelleBoissonFavorite;
}


// Services
void Humain::parle(const string texte) const
{
   cout << "(" << nom << ") -- " << texte << endl;
}

void Humain::sePresente() const
{
   cout << "(" << nom << ") -- " << "Bonjour, je suis " << this->getNom() << " et j'aime le " << this->getBoissonFavorite() << endl;
}

void Humain::boit() const
{
   cout << "(" << nom << ") -- " << "Ah ! un bon verre de " << getBoissonFavorite() << " ! GLOUPS !" << endl;
}

File: POO/Java/sequence5/src/western/Humain.java
package western;

public class Humain
{
    private String nom;
    private String boissonFavorite = "eau";

    public Humain()
    {

    }

    public Humain(String nom)
    {
        this.nom = nom;
    }

    public void setNom(String nom)
    {
        this.nom = nom;
    }

    public final String getNom()
    {
        return this.nom;
    }

    public void setBoissonFavorite(String boissonFavorite)
    {
        this.boissonFavorite = boissonFavorite;
    }

    public final String getBoissonFavorite()
    {
        return this.boissonFavorite;
    }

    public void sePresente()
    {
        String texte = "";
        texte = "Bonjour, je suis " + getNom() + " et j’aime le " + getBoissonFavorite();
        parle(texte);
    }

    public void boit()
    {
        String texte = "";
        texte = "Ah ! un bon verre de " + getBoissonFavorite() + " ! GLOUPS !";
        parle(texte);
    }

    public final void parle(String texte)
    {
        System.out.println("("+getNom()+") -- " + texte);
    }
}


